Core/Core/Views/GradeStatisticGraphView.swift

```diff
@@ -0,0 +1,140 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2020-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+
+
+public class GradeStatisticGraphView: UIView {
+    @IBOutlet weak var minPossibleLabel: UILabel!
+    @IBOutlet weak var maxPossibleLabel: UILabel!
+    @IBOutlet weak var averageLabel: UILabel!
+    @IBOutlet weak var minLabel: UILabel!
+    @IBOutlet weak var maxLabel: UILabel!
+    
+    @IBOutlet weak var minPossibleBar: UIView!
+    @IBOutlet weak var maxPossibleBar: UIView!
+    
+    @IBOutlet weak var graphArea: UIView!
+    
+    @IBOutlet weak var minConstraint: NSLayoutConstraint!
+    @IBOutlet weak var maxConstraint: NSLayoutConstraint!
+    @IBOutlet weak var yourScoreConstraint: NSLayoutConstraint!
+    @IBOutlet weak var meanConstraint: NSLayoutConstraint!
+    
+    @IBOutlet weak var minLabelXConstraint: NSLayoutConstraint!
+    @IBOutlet weak var maxLabelXConstraint: NSLayoutConstraint!
+
+    
+    @IBOutlet private var lines: [UIView]!
+    
+    @IBOutlet weak var yourScoreView: UIView!
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        loadFromXib()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        loadFromXib()
+        
+        yourScoreView.backgroundColor = Brand.shared.primary.ensureContrast(against: .named(.backgroundLightest))
+        yourScoreView.layer.cornerRadius = 8.0
+        for line in lines {
+            line.layer.cornerRadius = 1.0
+        }
+    }
+
+    public func update(_ assignment: Assignment) {
+        setupGraph(assignment: assignment)
+    }
+    
+    func setupGraph(assignment: Assignment) {
+        guard let stats = assignment.scoreStatistics, let points_possible = assignment.pointsPossible, let score = assignment.viewableScore, points_possible > 0 else {
+            isHidden = true
+            return
+        }
+        isHidden = false
+        let usableWidth = graphArea.frame.width - 48.0 - 2.0 // subtract 2 for half width of the outermost bars
```

Comment:
I haven't tried it yet but I suspect that this won't handle window size changes like when the user changes the width of the app when in side-by-side mode on the iPad. You might have to move the offset calculations and the constraint-constant-adjustments to `viewDidLayoutSubviews`. I could be wrong. It's possible `update(_:)` gets called at all the necessary times. I will try it out and see.

Reply:
I switched this to UIView's layoutSubviews (prior to calling `super.layoutSubviews()`) and the result is great. I even get this sweet resizing animation, with zero additional code:

![ezgif-3-f368d05a9ad2](https://user-images.githubusercontent.com/2116451/87622402-0dd98a80-c6f1-11ea-8f5d-4d3db9b6487e.gif)